### PR TITLE
Upgrade go dependencies

### PR DESCRIPTION
**Pull request by harryzcy**
Upgrade all dependencies in `go.mod`

`golang.org/x/net` v0.17.0 also fixes [CVE-2023-39325](https://github.com/advisories/GHSA-4374-p667-p6c8)

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -41,7 +41,7 @@ require (
 	github.com/go-ap/jsonld v0.0.0-20221030091449-f2a191312c73
 	github.com/go-chi/chi/v5 v5.0.10
 	github.com/go-chi/cors v1.2.1
-	github.com/go-co-op/gocron v1.31.1
+	github.com/go-co-op/gocron v1.35.2
```
Comment (delvh) [inline go.mod]: Did they fix their data race issue?

**Review by delvh — APPROVED** [review]

**Comment by silverwind** [comment]
Need to re-run tests a few times to confirm gocron is stable, see https://github.com/go-gitea/gitea/pull/27511

**Comment by silverwind** [comment]
gocron still causing the issue:

```
--- FAIL: TestAddTaskToScheduler (0.00s)
    tasks_test.go:39: 
        	Error Trace:	/home/runner/work/gitea/gitea/services/cron/tasks_test.go:39
        	Error:      	Not equal: 
        	            	expected: "30 5 4 * * *"
        	            	actual  : "5 4 * * *"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-30 5 4 * * *
        	            	+5 4 * * *
        	Test:       	TestAddTaskToScheduler
FAIL
```

File: go.mod (hunk 1 above)
Comment (silverwind) [reply go.mod]: Apparently not, see below.

Comment (delvh) [inline go.mod]: Apparently not.
```suggestion
	github.com/go-co-op/gocron v1.31.1
```

**Review by delvh — DISMISSED** [review]

**Review by silverwind — APPROVED** [review]

**Pull request merged**